eureka/S5_lightcurve_fitting/fitters.py

```diff
@@ -125,7 +170,428 @@ def demcfitter(time, data, model, uncertainty=None, verbose=True, **kwargs):
+
+    # Try to store each as an attribute
+    # for param in new_params:
+    #     setattr(params, param[0], param[1:])
+
+    # Make a new model instance
+    best_model = copy.copy(model)
+    best_model.components[0].update(fit_params, freenames)
+    # best_model.parameters = params
+    # best_model.name = ', '.join(['{}:{}'.format(k, round(v[0], 2)) for k, v in params.dict.items()])
+
+    import matplotlib.pyplot as plt
+
+    model.update(fit_params, freenames)
+    model_lc = model.eval()
+    residuals = (lc.flux - model_lc) #/ lc.unc
+
+    if meta.isplots_S5 > 1:
+        model.plot(time=lc.time, draw=True)
+
+        fig, ax = plt.subplots(2,1)
+        ax[0].errorbar(lc.time, lc.flux, yerr=lc.unc, fmt='.')
+        ax[0].plot(lc.time, model_lc, zorder = 10)
+        
+        ax[1].errorbar(lc.time, residuals, yerr=lc.unc, fmt='.')
+        
+        plt.savefig(meta.outputdir + 'figs/lc_emcee.png', dpi=300)
+        plt.show()
+
+    ln_like_val = (-0.5 * (np.sum((residuals / lc.unc) ** 2 + np.log(2.0 * np.pi * (lc.unc) ** 2))))
+    chi2 = np.sum((residuals / lc.unc) ** 2)
+    chi2red = chi2 / (len(lc.unc))
+    print(len(lc.unc))
+    print(chi2)
+    print(chi2red)
+    # best_model.parameters = params
+    # best_model.name = ', '.join(['{}:{}'.format(k, round(v[0], 2)) for k, v in params.dict.items()])
+
+    if meta.run_verbose:
+        for freenames_i, fit_params_i in zip(freenames, fit_params):
+            print('{0}: {1}'.format(freenames_i, fit_params_i))
+
+    if meta.isplots_S5 > 3:
+        rmsplot(lc, model_lc, meta, figname='allanplot_emcee.png')
+
+    return best_model
+
+
+def dynestyfitter(lc, model, meta, **kwargs):
+    """Perform sampling using emcee
+
+    Parameters
+    ----------
+    data: sequence
+        The observational data
+    model: ExoCTK.lightcurve_fitting.models.Model
+        The model to fit
+    uncertainty: np.ndarray (optional)
+        The uncertainty on the (same shape) data
+    method: str
+        The name of the method to use
+    name: str
+        A name for the best fit model
+    verbose: bool
+        Print some stuff
+
+    Returns
+    -------
+    lsq.Model.fit.fit_report
+        The results of the fit
+    """
+    # Concatenate the lists of parameters
+    #all_keys   = [i for j in [model.components[n].parameters.dict.keys()
+    #              for n in range(len(model.components))] for i in j]
+
+    # RUN LEAST SQUARES
+    lsq_sol = lsqfitter(lc, model, meta, **kwargs)
+    print(lsq_sol)
+    # SCALE UNCERTAINTIES WITH REDUCED CHI2 TODO: put a flag for that into config
+    lc.unc *= np.sqrt(lsq_sol.chi2red)
+
+    all_params = [i for j in [model.components[n].parameters.dict.items()
+                  for n in range(len(model.components))] for i in j]
+
+    #print(all_params)
+    # Group the different variable types
+    freenames = []
+    freepars = []
+    pmin = []
+    pmax = []
+    indep_vars = {}
+    for ii, item in enumerate(all_params):
```

Comment:
This for loop looks to be identical to the for loops used in other fitter functions.  If so, it should called as a function instead of repeating the code.

Reply:
Now addressed in 25_tjbEdits